design/cluster-scope-resource-filter.md

```diff
@@ -0,0 +1,502 @@
+	// +optional
+	// +nullable
+	IncludedResources []string `json:"includedResources,omitempty"`
+
+	// ExcludedResources is a slice of resource type names that are not
+	// included in the restore.
+	// Overwritten by ExcludedClusterScopeResources and 
+	// ExcludedNamespacedResources in conflict case in v1.10.
+	// Will be deprecated in v1.11.
+	// +optional
+	// +nullable
+	ExcludedResources []string `json:"excludedResources,omitempty"`
+
+	// IncludeClusterResources specifies whether cluster-scoped resources
+	// should be included for consideration in the restore. If null, defaults
+	// to true.
+	// +optional
+	// +nullable
+	IncludeClusterResources *bool `json:"includeClusterResources,omitempty"`
+
+	// IncludedClusterScopeResources is a slice of cluster scope 
+	// resource type names to include in the restore.
+	// If not specified, all cluster scope resources in the backup are included.
+	// Supersede IncludedResources in conflict case.
+	// +optional
+	// +nullable
+	IncludedClusterScopeResources []string `json:"includedClusterScopeResources,omitempty"`
+
+	// ExcludedClusterScopeResources is a slice of cluster scope 
+	// resource type names to exclude from the restore.
+	// Supersede ExcludedResources in conflict case.
+	// +optional
+	// +nullable
+	ExcludedClusterScopeResources []string `json:"excludedClusterScopeResources,omitempty"`
+
+	// IncludedNamespacedResources is a slice of namespace scope
+	// resource types names to include in the restore.
+	// If not specified, all namespaced scope resources in the backup are included.
+	// Supersede IncludedResources in conflict case.
+	// +optional
+	// +nullable
+	IncludedNamespacedResources []string `json:"includedNamespacedResources,omitempty"`
+
+	// ExcludedNamespacedResources is a slice of namespace scope
+	// resource type names to exclude from the backup.
+	// Supersede ExcludedResources in conflict case.
+	// +optional
+	// +nullable
+	ExcludedNamespacedResources []string `json:"excludedNamespacedResources,omitempty"`
+    ......
+}
+```
+
+## Alternatives Considered
+Proposal from Jibu Data [Issue 5120](https://github.com/vmware-tanzu/velero/issues/5120#issue-1304534563)
+
+
+## Security Considerations
+No security impact.
+
+## Compatibility
+### IncludedResources and ExcludedResources change
+To make the change backward compatible, `IncludedResources/ExcludedResources` parameters still apply to both cluster scope and namespace scope.
+After adding new parameters `IncludedClusterScopeResources`, `ExcludedClusterScopeResources`, `IncludedNamespacedResources` and `ExcludedNamespacedResources`, there could be conflict between the exiting and new parameters.
+When there is conflict, `IncludedClusterScopeResources/ExcludedClusterScopeResources` parameters have higher priority for cluster scope resources.
+`IncludedNamespacedResources/ExcludedNamespacedResources` parameters have higher priority for namespace scope resources.
+
+### IncludeClusterResources change
```

Comment:
See my comment above, IMO it would make it easier to understand if we deprecate this flag if `IncludedClusterScopeResources/ExcludedClusterScopeResources`  are introduced.
Otherwise, having both `IncludeClusterResources` and `IncludedClusterScopeResources` are confusing.

Reply:
We can't do that without losing an important use case. I agree that the naming is somewhat confusing, but we haven't been able to come up with a better name for the new IncludeClusterScopeResources parameter that is any clearer, and we can't change the old name without losing backwards compatibility. We just need to make sure that our documentation is very clear on the interaction of the parameters.